Add v5.4 to the DM overview documentation (#7563)
pingcap/docs: dm/dm-overview.md

## DM versions

> This document is dedicated to DM v5.4, the latest stable version of DM.

Author: ```suggestion
This document is for DM v5.4, the latest stable version of DM.
```
